app/views/students/attendances/index.html.erb

```diff
@@ -13,26 +13,29 @@
     <div class="level-item has-text-centered">
       <div>
         <p class="heading">Present</p>
-        <p class="title"><%= @attendance_record[true] %></p>
+        <p class="title"><%= @attendance_record[true] ? @attendance_record[true] : 0 %></p>
       </div>
     </div>
     <div class="level-item has-text-centered">
       <div>
         <p class="heading">Absent</p>
-        <p class="title"><%= @attendance_record[false]%></p>
+        <p class="title"><%= @attendance_record[false] ? @attendance_record[false] : 0 %></p>
       </div>
     </div>
     <div class="level-item has-text-centered">
       <div>
         <p class="heading">% of attendance</p>
-        <p class="title"><%= (@attendance_record[true].to_f / @attendance_record.values.sum) * 100 %></p>
+        <p class="title">
+          <%= (@attendance_record[true].to_f / @attendance_record.values.sum * 100).nan? ? 0 : (@attendance_record[true].to_f / @attendance_record.values.sum * 100) %> %
+        </p>
       </div>
     </div>
     <div class="level-item has-text-centered">
       <div>
         <p class="heading">% of absence</p>
-        <%# TODO: Fix NaN value when there is no attendances fields %>
-        <p class="title"><%= (@attendance_record[false].to_f / @attendance_record.values.sum) * 100 %></p>
+        <p class="title">
+          <%= (@attendance_record[false].to_f / @attendance_record.values.sum * 100).nan? ? 0 : (@attendance_record[false].to_f / @attendance_record.values.sum * 100) %> %
```

Comment:
same as above, you're doing the same operation twice, I think there's a better way to do this

Reply:
I apply a helper method to fix this.